api/py/ai/chronon/repo/dependency_tracker.py

```diff
@@ -0,0 +1,85 @@
+"""Object for checking the downstream dependency of an entity in the Chronon repository.
+"""
+
+#     Copyright (C) 2023 The Chronon Authors.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+import logging
+import os
+from ai.chronon.logger import get_logger
+from typing import List
+from ai.chronon.api.ttypes import \
+    GroupBy, Join
+from ai.chronon.repo import JOIN_FOLDER_NAME, \
+    GROUP_BY_FOLDER_NAME
+from ai.chronon.repo.validator import extract_json_confs
+
+
+class ChrononEntityDependencyTracker(object):
+    def __init__(self, chronon_root_path: str, log_level=logging.INFO):
+        self.logger = get_logger(log_level)
+        self.chronon_root_path = chronon_root_path
+
+    def check_join_downstream(self, conf_path) -> List[str]:
+        joins = extract_json_confs(
+            Join,
+            os.path.join(self.chronon_root_path, conf_path))
+        downstreams = []
+        if len(joins) == 0:
+            print(f"No joins found in {conf_path}")
+            return downstreams
+        elif len(joins) > 1:
+            print(f"Multiple joins found in {conf_path}")
```

Comment:
should we actually raise exception at line 42 and 39? similarly for line 61, 64?

Reply:
updated